controllers/Backend.php

```diff
@@ -1,104 +1,351 @@
+        $posts = $manager_article->get_post();
+        
+        $manager_user = new UserManager();
+        $admins = $manager_user->get_post_admin();
+        
+        }
+
+          if(filter_has_var(INPUT_POST, 'delete')){
+            $manager_article->delete_article_comments();
+            $manager_article->delete_article();
+        }
+
+        if(filter_has_var(INPUT_POST, 'submit')){
+            if(filter_has_var(INPUT_POST, 'title')){
+                $title = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'title')));
+            }
+            if(filter_has_var(INPUT_POST, 'chapo')){
+                $chapo = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'chapo')));
+            }
+            if(filter_has_var(INPUT_POST, 'content')){
+                $content = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'content')));
+            }
+            $posted = filter_input(INPUT_POST, 'public') ? "1" : "0";
+            
+            $errors = [];
+
+            if(empty($title) || empty($content) || empty($chapo)){
+                $errors['empty'] = "Veuillez remplir tous les champs svp";
+            }
+            if(strlen($title) < 5){
+                $errors['title'] = "Votre message doit contenir au moins 5 caractères.";
+            }
+            if(strlen($chapo) < 5){
+                $errors['chapo'] = "Votre chapô doit contenir au moins 5 caractères.";
+            }
+            if(strlen($content) < 5){
+                $errors['content'] = "Votre article doit contenir au moins 5 caractères.";
+            }
+
+            if(!empty($_FILES['image']['name'])){
+                $file = $_FILES['image']['name'];
+                $extensions = ['.png','.jpg','.jpeg','.gif','.PNG','.JPG','.JPEG','.GIF'];
+                $extension = strrchr($file,'.');
+  
+                if(!in_array($extension,$extensions)){
+                    $errors['image'] = "Cette image n'est pas valable.";
+                }
+            }
+            
+            if(empty($errors)){
+                $admin = $_SESSION['admin'];
+                $manager_article->edit($title,$chapo,$content,$posted, $admin, filter_input(INPUT_GET, 'id'));
+             
+                if(!empty($_FILES['image']['name']))
+                {
+                    $manager_article->update_img($_FILES['image']['tmp_name'], $extension);
+                    header("Location:/liste-de-tous-les-articles");
+                }
+                else
+                {
+                    header("Location:/liste-de-tous-les-articles");
+                }
+            }}else{
+                $errors = []; 
+            }
+        
+        if(!isset($_SESSION['admin'])){
+            header("location: /articles-a-la-une");
+        }
         
-        $page="backend/postback";
-        $topbar="backend/topbar";
-        require "views/layout.php";
+        if($posts == false){
+            header("Location:/page-erreur");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function header() is discouraged](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)